application/src/main/kotlin/net/corda/v5/application/messaging/FlowSession.kt

```diff
@@ -32,23 +33,13 @@ interface FlowSession {
     val counterparty: MemberX500Name
 
     /**
-     * Returns a [FlowInfo] object describing the flow [counterparty] is using. With [FlowInfo.flowVersion] it provides the necessary
-     * information needed for the evolution of flows and enabling backwards compatibility.
+     * Serializes and queues the given [payload] object for sending to the [counterparty]. Suspends until a response is
+     * received, which must be of the given [receiveType]. Remember that when receiving data from other parties the data
+     * should not be trusted until it's been thoroughly verified for consistency and that all expectations are
+     * satisfied, as a malicious peer may send you subtly corrupted data in order to exploit your code.
      *
-     * This method can be called before any send or receive has been done with [counterparty]. In such a case this will force them to start
-     * their flow.
-     */
-    @Suspendable
-    fun getCounterpartyFlowInfo(): FlowInfo
-
-    /**
-     * Serializes and queues the given [payload] object for sending to the [counterparty]. Suspends until a response is received, which must
-     * be of the given [receiveType]. Remember that when receiving data from other parties the data should not be trusted until it's been
-     * thoroughly verified for consistency and that all expectations are satisfied, as a malicious peer may send you subtly corrupted data
-     * in order to exploit your code.
-     *
-     * Note that this function is not just a simple send+receive pair: it is more efficient and more correct to use this when you expect to
-     * do a message swap than do use [send] and then [receive] in turn.
+     * Note that this function is not just a simple send+receive pair: it is more efficient and more correct to use this
+     * when you expect to do a message swap than do use [send] and then [receive] in turn.
```

Comment:
Needs `@param`s.

Should include an `@param` for `R`. E.g. `@param R The type to receive`.

Reply:
Here as well.